pyvista/core/filters/uniform_grid.py

```diff
@@ -750,11 +750,9 @@ def _check_fft_scalars(self):
             else:
                 raise MissingDataError('FFT filters require point scalars.')
 
-        scalars = self.point_data.active_scalars
-
-        meets_req = scalars.ndim == 2 and scalars.shape[1] == 2
-        if not meets_req:
+        if not np.issubdtype(self.point_data.active_scalars.dtype, np.complexfloating):
             raise ValueError(
                 'Active scalars must be complex data for this filter, represented '
-                'as an array with `dtype=numpy.complex128`.'
+                'as an array with a datatype of `numpy.complex64` or '
+                '`numpy.complex128`.'
             )
```

Comment:
There's a slight mismatch here related to a weird edge case: if someone passes an array with `np.complex256` dtype then they won't trip this error, but downstream operations will probably break (otherwise we'd support `np.complex256` as well).

But I remember noticing and not flagging this before, so it's possible that there's an earlier explicit filter for these two kinds of complex dtypes.

Reply:
It's simply not possible to add a `np.complex256` array to our datasets, we check that in:
https://github.com/pyvista/pyvista/blob/9a16343297b70e7d1722540ca0c968ede15f08c8/pyvista/core/datasetattributes.py#L822-L826